### PR TITLE
docs: added initial docz setup for documentation

### PR DESCRIPTION
**Pull request by anikethsaha**
- **What does this PR do?**
Added Docz documentation initial setup

- **Any background context you want to provide?**
https://github.com/devfolioco/react-otp-input/issues/36

- **Screenshots and/or Live Demo**
https://5d964f327aeaa4000813eff2--competent-ride-bdb7ab.netlify.com/docs-index

## Note
This is not yet completed. this is just the setup part.
I have deployed the site in netlify using this branch https://github.com/anikethsaha/react-otp-input/tree/docz/netlify-preview
One thing I faced while deploying was that we need to add the `.docz` folder which comes out when running `yarn docz:build` cause in netlify even after running the `npm run build && npm run docz:build` there, I guess some errors was there regarding the creation of `.docz/public` folder cause it was showing it's not able to locate it. *I am not even sure why its looking for that, our site is in `.docz/dist`*
 
Also do we need readme on the deployed docs ? https://5d964f327aeaa4000813eff2--competent-ride-bdb7ab.netlify.com/readme, 

*its looking good btw*

Any theme suggestions if you want?

### CODE REVIEW

**Comment by apollonian** [comment]
This looks amazing. Thanks @anikethsaha!

I guess we could have a demo page which consists of a playground and prop table. Something similar to https://gatsby-docz.netlify.com/docs/docs-button.

And another page containing the installation instructions. What do you think, @ajayns? 

Let me check if I get any errors while deploying.

**Comment by anikethsaha** [comment]
> I guess we could have a demo page which consists of a playground and prop table. Something similar to https://gatsby-docz.netlify.com/docs/docs-button.

Yeah surely it gonna be there. This is just the setup and for checking if setup is fine or not.
I guess there are other people too who wants to contribute in the docs issue. So for that only I didnt do that. We can split the work .

**Comment by apollonian** [comment]
@anikethsaha Yep. Let's keep it for a content for a separate PR. :v:

**Comment by anikethsaha** [comment]
cc @apollonian

File: package.json [hunk 1]
```diff
@@ -60,5 +62,8 @@
   },
   "bugs": {
     "url": "https://github.com/devfolioco/react-otp-input/issues"
+  },
+  "dependencies": {
+    "docz": "^2.0.0-rc.52"
```
Comment (apollonian) [inline package.json]: We should remove docz from the dependencies and move it devDependencies.

Comment (apollonian) [reply package.json]: Everything else looks good!

Comment (apollonian) [reply package.json]: Great work. 🙌

Comment (anikethsaha) [reply package.json]: done

**Pull request merged**